.github/workflows/publish_precompiled.yml

```diff
@@ -0,0 +1,70 @@
+name: Publish Precompiled
+
+on:
+  push:
+    branches:
+      - main
+
+env:
+  ELIXIR_OPTS: "--warnings-as-errors"
+  ERLC_OPTS: "warnings_as_errors"
+  LANG: C.UTF-8
+  S3_BUCKET: s3.hex.pm
+
+jobs:
+  publish_precompiled:
+    strategy:
+      fail-fast: true
+      # Prevent each job edit builds.txt concurrently.
+      max-parallel: 1
+      matrix:
+        include:
+          - otp: 24
+            otp_version: 24.3
+          - otp: 25
+            otp_version: 25.0
+    runs-on: ubuntu-18.04
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 50
+
+      - uses: erlef/setup-beam@v1
+        with:
+          otp-version: ${{ matrix.otp_version }}
+          version-type: strict
+
+      - name: Build Elixir Release
+        run: |
+          make Precompiled.zip
+          mv Precompiled.zip main-otp-${{ matrix.otp }}.zip
+          shasum -a 1 main-otp-${{ matrix.otp }}.zip > main-otp-${{ matrix.otp }}.zip.sha1sum
+          shasum -a 256 main-otp-${{ matrix.otp }}.zip > main-otp-${{ matrix.otp }}.zip.sha256sum
+
+      - name: Update builds.txt
+        run: |
+          date=$(date -u '+%Y-%m-%dT%H:%M:%SZ')
+          build_sha256=$(cut -d ' ' -f 1 main-otp-${{ matrix.otp }}.zip.sha256sum)
+
+          aws s3 cp s3://${{ env.S3_BUCKET }}/builds/elixir/builds.txt builds.txt || true
+          touch builds.txt
+          sed -i '/^main-otp-${{ matrix.otp }} /d' builds.txt
+          echo -e -n "main-otp-${{ matrix.otp }} ${{ github.sha }} ${date} ${build_sha256}\n$(cat builds.txt)" > builds.txt
+          sort -u -k1,1 -o builds.txt builds.txt
+        env:
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          AWS_DEFAULT_REGION: us-east-1
+
+      - name: Upload to S3
+        run: |
+          for file in main-otp-${{ matrix.otp }}.zip main-otp-${{ matrix.otp }}.zip.sha1sum main-otp-${{ matrix.otp }}.zip.sha256sum builds.txt
```

Comment:
I think you might have already asked about it, do we need to upload shasum files? They are redundant since we have builds.txt so maybe we don't?

Reply:
I’m not sure if we should keep it like release workflow does?